compiler/util/files.cpp

```diff
@@ -891,6 +891,14 @@ void codegen_makefile(fileinfo* mainfile, const char** tmpbinname,
 
   // List object files needed to compile this deliverable.
   fprintf(makefile.fptr, "CHPLUSEROBJ = \\\n");
+  if (!fLibraryCompile) {
```

Comment:
I guess this code here is to handle the dance you describe here:

> Beyond those changes to Makefile.exe, I also had to update the
compiler-generated Makefile to add the "main" CHPLSRC file to
CHPLUSROBJ in order to ensure that it still got built and linked in.
This required a bit of dancing to handle the fact that CHPLSRC
names a .c file for the C back-end, yet a .o file for the LLVM
back-end (?!)

The effect of this will be to either add:
`CHPLUSEROBJ = $(CHPLSRC:%.c=%)` or 
`CHPLUSEROBJ = $(CHPLSRC:%.o=%)`

to the Makefile, which I guess in `Makefile.exe` you then have `CHPLUSEROBJFILES = $(CHPLUSEROBJ:%=%.o)` which will have `CHPLUSEROBJFILES` rewrite it to have the `.o` extension.

Anyway, this seems pretty confusing to me, could you add a comment or two (or just linking it back to this issue) to explain this.